ratpack-test/src/test/groovy/ratpack/test/http/DefaultMultipartFormSpec.groovy

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ratpack.test.http
+
+import ratpack.test.http.internal.DefaultMultipartForm
+import spock.lang.Specification
+import spock.lang.Unroll
+
+/*
+    As defined in RFC 1867 and 2388 (see https://tools.ietf.org/html/rfc1867)
+ */
+
+class DefaultMultipartFormSpec extends Specification {
+
+    def 'form with single field'() {
+        given:
+            builder.field(field.key, field.value)
+
+        when:
+            def form = builder.build()
+
+        then:
+            form.headers == ['Content-Type': "multipart/form-data; boundary=${boundary}"]
+            form.body ==
+"""--${boundary}
+Content-Disposition: form-data; name="${field.key}"
+
+${field.value}
+--${boundary}--
+"""
+
+        where:
+            builder = DefaultMultipartForm.builder()
+            boundary = builder.boundary
+            field = ['name': 'Werner Heisenberg'].find { true }
```

Comment:
Would be more comprehensible as `['name': 'Werner Heisenberg'].entrySet().first()`, IMHO. Took me a while to understand the intention here.